entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationService.kt

```diff
@@ -31,6 +27,30 @@ class Neo4jAuthorizationService(
     override fun getSubjectGroups(sub: Option<Sub>): Set<URI> =
         neo4jAuthorizationRepository.getSubjectGroups(getSubjectUri(sub))
 
+    override fun getGroupsMemberships(
+        sub: Option<Sub>,
+        contextLink: String
+    ): Pair<Int, List<JsonLdEntity>> {
+
+        val groupsEntities = if (userIsAdmin(sub))
+            neo4jAuthorizationRepository.getGroupsAdmin(
+                neo4jAuthorizationRepository.getSubjectGroups((USER_PREFIX + sub.toStringValue()).toUri())
+            )
+        else
+            neo4jAuthorizationRepository
+                .getSubjectGroups((USER_PREFIX + sub.toStringValue()).toUri())
+                .map { neo4jAuthorizationRepository.getGroupEntity(it) }
```

Comment:
can't find a way to do a single query to retrieve all the groups?

Reply:
Yes it's possible but you didn't tell me that you wanted to keep the `getsubjectgroups` function?